appformer-provisioning/appformer-provisioning-client/src/main/java/org/kie/appformer/provisioning/client/deployment/DeploymentPopupViewImpl.java

```diff
@@ -0,0 +1,298 @@
+        this.portTextInput.setValue( port );
+    }
+
+    @Override
+    public String getManagementPort( ) {
+        return managementPortTextInput.getValue();
+    }
+
+    @Override
+    public void setManagementPort( String managementPort ) {
+        this.managementPortTextInput.setValue( managementPort );
+    }
+
+    @Override
+    public String getManagementUser( ) {
+        return managementUserTextInput.getValue();
+    }
+
+    @Override
+    public void setManagementUser( String managementUser ) {
+        this.managementUserTextInput.setValue( managementUser );
+    }
+
+    @Override
+    public String getManagementPassword( ) {
+        return managementPasswordTextInput.getValue();
+    }
+
+    @Override
+    public void setManagementPassword( String managementPassword ) {
+        this.managementPasswordTextInput.setValue( managementPassword );
+    }
+
+    @Override
+    public void loadDataSourceOptions( final List<Pair<String, String>> dataSourceOptions, final boolean addEmptyOption ) {
+        dataSourceSelector.clear();
+        if ( addEmptyOption ) {
+            dataSourceSelector.add( newOption( "", "" ) );
+        }
+        for ( Pair<String, String> optionPair: dataSourceOptions ) {
+            dataSourceSelector.add( newOption( optionPair.getK1(), optionPair.getK2() ));
+        }
+        refreshDataSourceSelector();
+    }
+
+    @Override
+    public String getDataSource( ) {
+        return dataSourceSelector.getValue();
+    }
+
+    @Override
+    public void setDataSource( String dataSource ) {
+        dataSourceSelector.setValue( dataSource );
+        refreshDataSourceSelector();
+    }
+
+    private Option newOption( final String text, final String value ) {
+        final Option option = new Option();
+        option.setValue( value );
+        option.setText( text );
+        return option;
+    }
+
+    private void refreshDataSourceSelector() {
+        Scheduler.get().scheduleDeferred( new Scheduler.ScheduledCommand() {
+            @Override
+            public void execute() {
+                dataSourceSelector.refresh();
+            }
+        } );
+    }
+
+    @Override
+    public void show() {
+        modal.show();
+    }
+
+    @Override
+    public void hide() {
+        modal.hide();
+    }
+
+    @Override
+    public void showMessage( String title, String message ) {
+        YesNoCancelPopup.newYesNoCancelPopup( title, message, null, null, ( org.uberfire.mvp.Command ) ( ) -> {
+            //do nothing.
+        } ).show();
+    }
+
+    @EventHandler( "local-server-radio-input" )
+    private void onLocalServerRadioChange( final ChangeEvent event ) {
+        presenter.onLocalChange();
+    }
+
+    @EventHandler( "remote-server-radio-input" )
+    private void onRemoteServerRadioChange( final ChangeEvent event ) {
+        presenter.onRemoteChange();
+    }
+
+    @EventHandler( "test-connection-button" )
+    private void onTestConnection( final ClickEvent event ) {
```

Comment:
This can be made native by changing the parameter to `@ForEvent( "click" ) Event event`.